filament/include/filament/TransformManager.h

```diff
@@ -222,6 +222,23 @@ class UTILS_PUBLIC TransformManager : public FilamentAPI {
      */
     children_iterator getChildrenEnd(Instance parent) const noexcept;
 
+    /**
+     * Sets a scale of a transform component.
+     * This scale is preserved for sign check when decomposing transform matrix.
+     * @param ci     The instance of the transform component to set the scale to.
+     * @param scale  The preserved scale.
+     * @see getScale()
+     */
+    void setScale(Instance ci, const math::float3& scale) noexcept;
```

Comment:
I'll let @pixelflinger do the full review, but I'm pretty sure we do not want something like this on `TransformManager`.